api/handlers/bitcoin/bitcoin_handler.go

```diff
@@ -119,14 +119,14 @@ func Update(c *fiber.Ctx) error {
 		return c.Status(badReq.Status).JSON(err)
 	}
 
-	node := c.Locals("node").(*bitcoinv1alpha1.Node)
+	node := c.Locals("node").(bitcoinv1alpha1.Node)
 
-	node, err := service.Update(dto, node)
+	updated, err := service.Update(dto, &node)
```

Comment:
@mFarghaly 
as you can see we have to &protocol coz XYZ.service uses  pointer to protocol